pkg/protocol/recordlayer/header.go

```diff
@@ -17,11 +17,14 @@ type Header struct {
 	Version        protocol.Version
 	Epoch          uint16
 	SequenceNumber uint64 // uint48 in spec
+
+	// Optional Fields
+	ConnectionID []byte
 }
 
 // RecordLayer enums
 const (
-	HeaderSize        = 13
+	DefaultHeaderSize = 13
```

Comment:
Instead of `Default` it would be nice to hint at CID somehow.

`HeaderSizePreCID` might not be the right name.

Reply:
@Sean-Der sounds good -- I was actually thinking of just keeping this as `HeaderSize` when I finish out this PR as this is a public `const` so any change is technically a breaking change. What do you think?

Reply:
@hasheddan either works for me!

ConnectionID warrants a `V3`. It will be easier to get excitement/have people see the feature that way. So feel free to break the API in all the ways that make sense :)

I hope we can merge to master and tag `v3.0.0-alpha.1` very soon then!

Reply:
Went with `FixedHeaderSize` as it feels like it could age a little better since it isn't directly coupled to the implementation of connection IDs (i.e. in theory there could be another extension in the future that modified header structure).